pkg/daemon/client/client.go

```diff
@@ -88,72 +88,90 @@ func (c *Client) Run(ctx context.Context, r *RunRequest) (io.ReadCloser, error)
 	return c.request(ctx, "POST", "/run", bytes.NewReader(body.Bytes()))
 }
 
-// ProcessBuildResponse parses a response from a `build` call
-func ProcessBuildResponse(r io.ReadCloser) (*api.BuildOutput, error) {
+func processGeneric(r io.ReadCloser, fnProgress, fnResult func(interface{}) error) error {
 	var msg tgwriter.Msg
 
 	for dec := json.NewDecoder(r); ; {
 		err := dec.Decode(&msg)
 		if err != nil {
-			return nil, err
+			return err
 		}
 
 		switch msg.Type {
 		case "progress":
-			m, err := base64.StdEncoding.DecodeString(msg.Payload.(string))
+			err = fnProgress(msg.Payload)
 			if err != nil {
-				return nil, err
+				return err
 			}
 
-			fmt.Print(string(m))
-
 		case "error":
-			return nil, errors.New(msg.Error.Message)
+			return errors.New(msg.Error.Message)
 
 		case "result":
-			resp := &BuildResponse{}
-			err := mapstructure.Decode(msg.Payload, resp)
-			if err != nil {
-				return nil, err
-			}
-
-			return resp, nil
+			return fnResult(msg.Payload)
 
 		default:
-			return nil, errors.New("unknown message type")
+			return errors.New("unknown message type")
 		}
 	}
 }
 
-// ProcessDescribeResponse parses a response from a `describe` call
-func ProcessDescribeResponse(r io.ReadCloser) error {
-	var msg tgwriter.Msg
+func printProgress(progress interface{}) error {
+	m, err := base64.StdEncoding.DecodeString(progress.(string))
+	if err != nil {
+		return err
+	}
 
-	for dec := json.NewDecoder(r); ; {
-		err := dec.Decode(&msg)
-		if err != nil {
-			return err
-		}
+	fmt.Print(string(m))
+	return nil
+}
 
-		switch msg.Type {
-		case "progress":
-			m, err := base64.StdEncoding.DecodeString(msg.Payload.(string))
-			if err != nil {
-				return err
-			}
+// ProcessRunResponse parses a response from a `run` call
+func ProcessRunResponse(r io.ReadCloser) error {
+	return processGeneric(
+		r,
+		printProgress,
+		func(result interface{}) error {
+			return nil
+		},
+	)
+}
 
-			fmt.Print(string(m))
+// ProcessListResponse parses a response from a `list` call
+func ProcessListResponse(r io.ReadCloser) error {
+	return processGeneric(
+		r,
+		printProgress,
+		func(result interface{}) error {
+			return nil
+		},
+	)
+}
 
-		case "error":
-			return errors.New(msg.Error.Message)
+// ProcessBuildResponse parses a response from a `build` call
+func ProcessBuildResponse(r io.ReadCloser) (*api.BuildOutput, error) {
+	var resp *BuildResponse
+	err := processGeneric(
+		r,
+		printProgress,
+		func(result interface{}) error {
+			resp = &BuildResponse{}
+			return mapstructure.Decode(result, resp)
+		},
+	)
+
+	return resp, err
+}
 
-		case "result":
+// ProcessDescribeResponse parses a response from a `describe` call
+func ProcessDescribeResponse(r io.ReadCloser) error {
```

Comment:
so call it `parse` instead of `process`? (This might be an artefact from a direct Portuguese -> English translation)

Reply:
Doone!